### ISSUE TITLE
Switching between list types is not possible anymore

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/text#3256]
When having the cursor set in a list it is no longer possible to switch to a different type, e.g. from bullet list to checkbox

**Comment by max-nextcloud** [comment]
Switching between bullet and numbered list seems to work fine for me.
"Only" issue seems to be switching between checkbox and the other two list types in either way.
When on a checkbox list the other two list types are greyed out. So looks like tiptap already thinks it cannot do this.

I don't remember how this was fixed after the tiptap2 migration but i remember discussing it at length in a tiptap issue: https://github.com/ueberdosis/tiptap/issues/2428

**Comment by juliushaertl** [comment]
I looked a bit into this when checking other list issues:

It seems that this part here is failing to validate that the list items are valid content for the different types bullet and task list for example, as task list uses a different taskitem node type: https://github.com/ueberdosis/tiptap/blob/d6657233e718ee465dbe096c8f40a7910d15e696/packages/core/src/commands/toggleList.ts#L98

We or tiptap either would need to switch to a shared item node type or maybe could rework the command to not only change the list type but also the list item types. Haven't looked further on what would be the best or lowest effort approach.

**Comment by jospoortvliet** [comment]
> The problem seems that some commands are reported as not possible by `$editor.can()` even though they can be used without problems. E.g. switching from a bullet list to either of ordered list (Ctrl-Shift-7), check list (Ctrl-Shift-9) or blockquote (Ctrl-Shift-B) works well via keyboard shortcuts, even though e.g. `$editor.can().toggleOrderedList()` returns false, leading to the corresponding menu action being disabled.
> 
> Related upstream issue: [ueberdosis/tiptap#4057](https://github.com/ueberdosis/tiptap/issues/4057)

So it is possible to do it with a shortcut but the menu can't do it? That's very confusing 🥲 

<img width="222" height="291" alt="Image" src="https://github.com/user-attachments/assets/3d4c600a-b355-48df-bc56-214dcd685b04" />

It's a bit of a frequent annoyance so I'm going to try to learn the shortcuts then, I'm very happy with the shortcut overview in that regard. Makes it easy.

**Comment by janbaum** [comment]
Just saw this, weeks after creating #8112. I think my issue is kind of a duplicate of this one, so I will comment here, feel free to close #8112 if you agree 👍 Also, I don't fully understand the previous analysis you all did on this issue, so I thought I just describe what I see and you can happily ignore it, if says what you figured out previously 😬 

First, switching between all types of lists works, **if**:
1. you switch off the current type
2. open the menu again
3. select the desired type

There seems no option to changes list types directly, no matter from which/to which type. As said by you before, shortcuts work fine.

I think this is a rather famous papercut among users, too :)

**Comment by mejo-** [comment]
Crazy news, but apparently this got fixed 🎉 Most likely it happened with the upgrade to Tiptap v3. I can no longer reproduce this issue in Nextcloud 33 and above.

**Comment by janbaum** [comment]
> Crazy news, but apparently this got fixed 🎉 Most likely it happened with the upgrade to Tiptap v3. I can no longer reproduce this issue in Nextcloud 33 and above.

Yes, I can now directly change between ordered-, and unordered lists :) Although changing from (un)ordered list to task list still requires to first disable the previous type and then manually select task list

**Comment by mejo-** [comment]
> Although changing from (un)ordered list to task list still requires to first disable the previous type and then manually select task list

This will be fixed with the upcoming server releases as well, see #8536
